docs/csharp/how-to/search-strings.md

```diff
@@ -54,13 +54,14 @@ The following code uses regular expressions to validate the format of each strin
 
 This single search pattern matches many valid strings. Regular expressions are better to search for or validate against a pattern, rather than a single text string.
 
-You can try these samples by looking at the code in our [GitHub repository](https://github.com/dotnet/docs/tree/master/samples/snippets/csharp/how-to/strings). Or, you can cownload the samples [as a zip file](https://github.com/dotnet/docs/tree/master/samples/snippets/csharp/how-to/strings.zip)
+You can try these samples by looking at the code in our [GitHub repository](https://github.com/dotnet/docs/tree/master/samples/snippets/csharp/how-to/strings). Or you can download the samples [as a zip file](https://github.com/dotnet/docs/tree/master/samples/snippets/csharp/how-to/strings.zip).
 
 ## See Also  
 
  [C# Programming Guide](../programming-guide/index.md)  
  [Strings](../programming-guide/strings/index.md)  
- [LINQ and Strings](http://msdn.microsoft.com/library/6c34169f-7a39-436a-98d8-9a7283043942)   
+ [LINQ and Strings](../programming-guide/concepts/linq/linq-and-strings)   
  <xref:System.Text.RegularExpressions.Regex?displayProperty=nameWithType>     
- [.NET Framework Regular Expressions](https://msdn.microsoft.com/library/hs600312)   
- [Regular Expression Language - Quick Reference](http://msdn.microsoft.com/library/930653a6-95d2-4697-9d5a-52d11bb6fd4c)   
+ [.NET Framework Regular Expressions](../../standard/base-types/regular-expressions)   
+ [Regular Expression Language - Quick Reference](../../standard/base-types/regular-expression-language-quick-reference)   
+ [Best practices for using strings in .NET](../../standard/base-types/best-practices-strings)  
```

Comment:
@BillWagner I'm afraid all those inside-repo links require `.md` extension suffix.

Reply:
thanks. On it.